audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/AuditServiceAsyncProxy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.audit.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.palisade.service.audit.model.AuditErrorMessage;
+import uk.gov.gchq.palisade.service.audit.model.AuditMessage;
+import uk.gov.gchq.palisade.service.audit.model.AuditSuccessMessage;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * An asynchronous service for processing requests that might be slow.
+ */
+public class AuditServiceAsyncProxy {
```

Comment:
Would have expected to see a unit test for this class.

Reply:
Could add a unit test class to test each message type is dealt with properly.

Reply:
Yes that was what I was thinking of.   A set of tests that would test out each of the nooks and crannies in the audit method.

Reply:
Unit tests added.